wordpress-mobile/WordPress-Android
# [Modal Layout Picker] Retry view

**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2740

## Description
Adds a retry view to handle the case where the loading of layouts fails and there are no cached data

## To test:
Layout Picker should show [when creating a new page from *My Site* or *Site Pages* the *Modal Layout Picker* appears](https://github.com/wordpress-mobile/WordPress-Android/pull/12482).

1. Fresh install the app or clear storage (Device Settings>Apps>WordPress Beta>Storage & cache>Clear storage)
1. Login
1. **Disable the internet** connection (e.g. flight mode)
1. Open the Modal Layout Picker 
1. Verify that you see the **error/retry screen** (check screenshot below)
1. Press the *Retry* button
1. Verify that nothing happens
1. **Enable the internet** connection
1. Press the **Retry** button
1. Verify that the layouts did load

## Screenshot

<img width="360" src="https://user-images.githubusercontent.com/304044/96564056-59d93400-12cb-11eb-8920-c9cd9aa828d5.png"/>

PR submission checklist:
